synthetics-client/scheduler.py

```diff
@@ -0,0 +1,149 @@
+                # response status
+                response_status = suites["response"]["status"]
+                if (not all((type(x) is int) for x in suites["response"]["status"])): 
+                    raise Exception("Response Status not accepted")
+
+                # method
+                method = suites["request"]["method"]
+                if (method not in ["GET","POST"]): raise Exception("Request Method not accepted")
+
+                headers = suites["request"]["headers"]
+                if (type(headers) != dict): raise Exception("Request Headers not suitable")
+
+                req_json = suites["request"]["json"]
+                if (type(req_json) != dict): raise Exception("Request JSON not suitable")
+
+                # type of schedule, like interval or cron
+                schedule_type = suites["scheduler"]["schedule-type"]
+                if (schedule_type not in ["interval","cron"]): raise Exception("Schedule type not supported")
+                
+                interval_job = {}
+                cron_job = {}
+                if (schedule_type == 'interval'):
+                    unit = suites["scheduler"]["schedule"]["unit"]
+                    interval = int(suites["scheduler"]["schedule"]["period"])
+                    interval_job = {unit: interval}
+                elif (schedule_type == 'cron'):
+                    cron_job = suites["scheduler"]["schedule"]
+                
+                # start trying to run the jobs
+                try:
+                    print(filename + " succ")
+                    
+                    # Goes over every host found in a suite file
+                    for host in suites["hosts"]:
+                        if DEBUG: print("Start pinging for: " + host)
+                        # TODO: send information in through an object instead of multiple arguments
+                        p = Ping(self.client, host, suite_id, suites["name"], response_status, method, headers, req_json)
+                        # one ping to call immediately and the job will only run after a initial interval passes
+                        if schedule_type == 'interval': p.ping()
+                        job_process = scheduler.add_job(p.ping, schedule_type, **cron_job, **interval_job, replace_existing=True)
+                        suite_id += 1
+                    if DEBUG: print()
+                except Exception as e:
+                    print("job runtime error: " + str(e))
+            except Exception as e:
+                print(filename + " not formatted correctly: " + str(e))
+
+        # have to keep program running for scheduler to work, stops when the number of jobs is 0
+        while len(scheduler.get_jobs()) > 0:
+            time.sleep(2)
+
+    def create_synthetics_index(self):
+        # Try looking for 'synthetics-logs' index, will make one if not found
+        response = ''
+        try:
+            response = self.client.search(index='synthetics-logs')
```

Comment:
Use complex name for system indices. Taking indices from other plugins for example.

`.opensearch-observability-object`